include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +64,94 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v, int = -1)
+    {
+        out << v;
+    }
+};
+
+template <>
+struct output_formatter<std::string>
+{
+    template <typename T>
+    static void append(std::string& out, T const& val, int significant_digits = -1)
+    {
+        BOOST_STATIC_ASSERT((boost::is_arithmetic<T>::value));
+        std::stringstream ss;
+        if(std::numeric_limits<T>::is_specialized)
+        {
+            if(significant_digits < 0)
+                ss.precision(std::numeric_limits<T>::digits10);
+            else
+                ss.precision(significant_digits);
+        }
+        ss << val;
+        out += ss.str();
+    }
+
+    static void append(std::string& out, char c, int = -1)
+    {
+        out += c;
+    }
+
+    static void append(std::string& out, char const* s, int = -1)
+    {
+        out += s;
+    }
+};
+
+template <>
+struct output_formatter<std::ostream>
+{
+    template <typename T>
+    static void append(std::ostream& out, T const& val, int significant_digits = -1)
+    {
+        BOOST_STATIC_ASSERT((boost::is_arithmetic<T>::value));
+        if(std::numeric_limits<T>::is_specialized)
+        {
+            if(significant_digits < 0)
```

Comment:
a space after `if` is needed, please also use brackets for `if...else`

Reply:
there are other cases of missing spaces after `if`, so please address them as well